### ISSUE TITLE
Adjust dependencies versions according to older qiskit-terra versions

### ISSUE THREAD

**Issue by karlaspuldaro** [qiskit-community/submit-metriq#7]
The benchmark task code behaves differently depending on the qiskit version being used.

<details><summary>Some output examples</summary>
<pre>
Qiskit version:  0.13.0
Sample size:  100
Architecture: ibm_rochester
Circuit depth ave: 24.81 | Gate count ave: 49.76

Architecture: rigetti_16q_aspen
Circuit depth ave: 37.0 | Gate count ave: 80.56

---
Qiskit version:  0.14.0
Sample size:  100
Architecture: ibm_rochester
Circuit depth ave: 13.4 | Gate count ave: 21.52

Architecture: rigetti_16q_aspen
Circuit depth ave: 17.56 | Gate count ave: 29.82

---
Qiskit version:  0.15.0
Sample size:  100
/Users/kspuldaro/github/submit-metriq/.tox/python3.7/lib/python3.7/site-packages/qiskit/circuit/register.py:121: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
  if not isinstance(key, (int, np.int, np.int32, np.int64, slice, list)):

(The warning above is printed multiple times until the end of runtime...)

---
Qiskit version:  0.23.0
Sample size:  100 (works well with small sample sizes... eg. 5)
Architecture: ibm_rochester
Traceback (most recent call last):
  File "scripts/circuit_depth_and_gate_count.py", line 45, in <module>
    result = transpile(circuit, coupling_map=coupling_map, optimization_level=3)
  File "/Users/kspuldaro/opt/anaconda3/envs/metriq/lib/python3.7/site-packages/qiskit/compiler/transpiler.py", line 388, in transpile
    transpile_config["pass_manager_config"].backend_properties,
  File "/Users/kspuldaro/opt/anaconda3/envs/metriq/lib/python3.7/site-packages/qiskit/compiler/transpiler.py", line 474, in _serial_transpile_circuit
    result = pass_manager.run(circuit, callback=callback, output_name=output_name)
  File "/Users/kspuldaro/opt/anaconda3/envs/metriq/lib/python3.7/site-packages/qiskit/transpiler/passmanager.py", line 528, in run
    return super().run(circuits, output_name, callback)
  File "/Users/kspuldaro/opt/anaconda3/envs/metriq/lib/python3.7/site-packages/qiskit/transpiler/passmanager.py", line 228, in run
    return self._run_single_circuit(circuits, output_name, callback)
  File "/Users/kspuldaro/opt/anaconda3/envs/metriq/lib/python3.7/site-packages/qiskit/transpiler/passmanager.py", line 283, in _run_single_circuit
    result = running_passmanager.run(circuit, output_name=output_name, callback=callback)
  File "/Users/kspuldaro/opt/anaconda3/envs/metriq/lib/python3.7/site-packages/qiskit/transpiler/runningpassmanager.py", line 125, in run
    dag = self._do_pass(pass_, dag, passset.options)
  File "/Users/kspuldaro/opt/anaconda3/envs/metriq/lib/python3.7/site-packages/qiskit/transpiler/runningpassmanager.py", line 173, in _do_pass
    dag = self._run_this_pass(pass_, dag)
  File "/Users/kspuldaro/opt/anaconda3/envs/metriq/lib/python3.7/site-packages/qiskit/transpiler/runningpassmanager.py", line 202, in _run_this_pass
    new_dag = pass_.run(dag)
  File "/Users/kspuldaro/opt/anaconda3/envs/metriq/lib/python3.7/site-packages/qiskit/transpiler/passes/utils/gate_direction.py", line 299, in run
    return self._run_coupling_map(dag, layout_map)
  File "/Users/kspuldaro/opt/anaconda3/envs/metriq/lib/python3.7/site-packages/qiskit/transpiler/passes/utils/gate_direction.py", line 177, in _run_coupling_map
    f"Flipping of gate direction is only supported "
qiskit.transpiler.exceptions.TranspilerError: "Flipping of gate direction is only supported for ['cx', 'cz', 'ecr'] at this time, not 'swap'."
</pre>
</details>

These might occur due to incompatible qiskit dependency versions.

**Comment by 1ucian0** [comment]
A possible way to do this is with env-specific and multi-dimentional configuration (this second part, addressing the problem of CLI for specific terra versions)

```
[tox]

envlist = {python3.7, python3.8}-{terra0.23,terra0.24}
minversion = 3.7

[testenv]
description = run benchmark task file
deps =
    terra0.23: qiskit-terra==0.23.3
    terra0.24: qiskit-terra==0.24.0
    requests
    pyzx
commands =
    python {toxinidir}/scripts/circuit_depth_and_gate_count.py
```

With `terra0.24: <something>` you can specify a dependency that only applies to that environment. Then, you can call that environment like this:

`tox -e python3.7-terra0.23`

**Comment by karlaspuldaro** [comment]
Known `qiskit-terra` versions that fail executing the task:
* 0.13.0, 0.14.2: `AttributeError: module 'numpy' has no attribute 'float'.` - Use NumPy < 1.20
* 0.15.2: `AttributeError: module 'numpy' has no attribute 'int'.` - Use NumPy < 1.20

~~0.19.2, 0.20.2, 0.21.2, 0.22.4, 0.23.3~~: Fixed by #11
